pkg/api/v20191231preview/openshiftcluster_validatestatic.go

```diff
@@ -198,6 +197,13 @@ func (sv openShiftClusterStaticValidator) validateNetworkProfile(path string, np
 		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, path+".podCidr", "The provided pod CIDR '%s' is invalid: must be IPv4.", np.PodCIDR)
 	}
 
+	for _, s := range api.JoinCIDRRange {
+		_, cidr, _ := net.ParseCIDR(s)
+		if cidr.Contains(pod.IP) {
```

Comment:
I don't think this works.
For example, when they set `100.0.0.0/8` as podCidr, `100.0.0.1`, which is the first IP of it, is not contained by any `JoinCIDRRange`, but it should be rejected because `100.0.0.0/8` overlaps with (completely includes) `100.64.0.0/16`.

we need the both-way check.

Reply:
I modified the condition for both-way check, however add condition to skip 0.0.0.0/0 also

Reply:
Why do we have to skip 0.0.0.0/0? Should 0.0.0.0/0 be allowed?
If it's because of the test failures, we have to change the existing test cases instead.

https://github.com/Azure/ARO-RP/blob/a69d5854e4f248c1e7b224330a4cb5ee5d6744f7/pkg/api/v20191231preview/openshiftcluster_validatestatic_test.go#L828-L836

Reply:
Why do we skip 0.0.0.0/0? Should it be allowed?
If we skip it because of the test failures, the tests are something that should be fixed.

https://github.com/Azure/ARO-RP/blob/a69d5854e4f248c1e7b224330a4cb5ee5d6744f7/pkg/api/v20191231preview/openshiftcluster_validatestatic_test.go#L827-L836

It would be better to change this 0.0.0.0/0 to other cidr that should be allowed so that we can check `PropertyChangeNotAllowed`.